packages/domains-table/src/utils/resolve-domain-status.tsx

```diff
@@ -0,0 +1,674 @@
+import { Button } from '@automattic/components';
+import { localizeUrl } from '@automattic/i18n-utils';
+import moment from 'moment';
+import {
+	transferStatus,
+	type as domainTypes,
+	gdprConsentStatus,
+	useMyDomainInputMode,
+} from './constants';
+import { isExpiringSoon } from './is-expiring-soon';
+import { isRecentlyRegistered } from './is-recently-registered';
+import {
+	domainManagementEdit,
+	domainManagementEditContactInfo,
+	domainMappingSetup,
+	domainUseMyDomain,
+} from './paths';
+import {
+	SETTING_PRIMARY_DOMAIN,
+	INCOMING_DOMAIN_TRANSFER_STATUSES_IN_PROGRESS,
+	GDPR_POLICIES,
+	DOMAIN_EXPIRATION,
+} from './url-support';
+import type { ResponseDomain } from './types';
+import type { I18N, TranslateResult } from 'i18n-calypso';
+
+type ResolveDomainStatusReturn =
+	| {
+			statusText: TranslateResult | TranslateResult[];
+			statusClass:
+				| 'status-error'
+				| 'status-warning'
+				| 'status-alert'
+				| 'status-success'
+				| 'status-neutral'
+				| 'status-premium';
+			status: TranslateResult;
+			icon: 'info' | 'verifying' | 'check_circle' | 'cached' | 'cloud_upload' | 'download_done';
+			listStatusWeight?: number;
+			noticeText?: TranslateResult | Array< TranslateResult > | null;
+	  }
+	| Record< string, never >;
+
+export type ResolveDomainStatusOptionsBag = {
+	isJetpackSite?: boolean | null;
+	isSiteAutomatedTransfer?: boolean | null;
+	isDomainOnlySite?: boolean | null;
+	siteSlug?: string | null;
+	currentRoute?: string | null;
+	getMappingErrors?: boolean | null;
+};
+
+export type DomainStatusPurchaseActions< T > = {
```

Comment:
I do not have access to the `Purchase` type here so I made this generic so it can be set in the client